src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -0,0 +1,165 @@
+"""
+File       : MSPileupObj.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileupObj module provides MSPileup data structure:
+
+{
+    "pileupName": string with the pileup dataset
+    "pileupType": string with a constant value
+    "insertTime": int, seconds since epoch in GMT timezone
+    "lastUpdateTime": int, seconds since epoch in GMT timezone
+    "expectedRSEs": ["Disk1", "Disk2", etc],  # these values need to be validated against Rucio
+    "currentRSEs": ["Disk1", "Disk3"],  # values provided by the MS itself
+    "fullReplicas": integer,  # total number of replicas to keep on Disk
+    "campaigns": [ "name", ... ] # list of workflow campaigns using this pileup
+    "containerFraction": real number with the container fraction to be distributed (TBFD)
+    "replicationGrouping": string with a constant value (DATASET or ALL, to be in sync with Rucio)
+    "activatedOn": int, seconds since epoch in GMT timezone
+    "deactivatedOn": int, seconds since epoch in GMT timezone
+    "active": boolean,
+    "pileupSizeGB": integer, current size of the pileup in GB (integer)
+    "rulesList: list of strings (rules) used to lock the pileup id
+}
+
+The data flow should be done via list of objects, e.g.
+[<pileup object 1>, <pileup object 2>, ..., <pileup object n>]
+"""
+
+# system modules
+import json
+
+# WMCore modules
+from WMCore.MicroService.Tools.Common import getMSLogger
+from WMCore.Lexicon import dataset
+from Utils.Timers import gmtimeSeconds
+
+
+class MSPileupObj(object):
+    """
+    MSPileupObj defines MSPileup data stucture
+    """
+    def __init__(self, pdict, verbose=None, logger=None, validRSEs=None):
+        self.logger = getMSLogger(verbose, logger)
+        if not validRSEs:
+            validRSEs = []
+        self.validRSEs = validRSEs
+
+        self.data = {
+            'pileupName': pdict.get('pileupName', ''),
+            'pileupType': pdict.get('pileupType', 'classic'),
```

Comment:
I think the pileupType default value should be `""`. Thus ensuring that the end-user provides the appropriate configuration.

Reply:
done